pkg/deploy/webhook_chart.go

```diff
@@ -1,61 +1,122 @@
-	ErrFoundZeroWatchers             = errors.New("found 0 watcher resources, expected at least 1")
+	ErrSKRWebhookNotInstalled         = errors.New("skr webhook resources are not installed")
+	ErrSKRWebhookWasNotRemoved        = errors.New("installed skr webhook resources were not removed")
+	ErrLoadBalancerIPIsNotAssigned    = errors.New("load balancer service external ip is not assigned")
+	ErrCABundleAndCaCertMismatchFound = errors.New("found CABundle and CA certificate mismatch")
 )
 
 type WatchableConfig struct {
 	Labels     map[string]string `json:"labels"`
 	StatusOnly bool              `json:"statusOnly"`
 }
 
-func ResolveSKRChartResourceName(resourceNameTpl string) string {
-	return fmt.Sprintf(resourceNameTpl, ReleaseName)
+func generateWatchableConfigs(watchers []v1alpha1.Watcher) map[string]WatchableConfig {
+	chartCfg := make(map[string]WatchableConfig, 0)
+	for _, watcher := range watchers {
+		statusOnly := watcher.Spec.Field == v1alpha1.StatusField
+		chartCfg[watcher.GetModuleName()] = WatchableConfig{
+			Labels:     watcher.Spec.LabelsToWatch,
+			StatusOnly: statusOnly,
+		}
+	}
+	return chartCfg
+}
+
+func ResolveSKRChartResourceName(resourceNameTpl string, kymaObjKey client.ObjectKey) string {
+	return fmt.Sprintf(resourceNameTpl, skrChartReleaseName(kymaObjKey))
+}
+
+func skrChartReleaseName(kymaObjKey client.ObjectKey) string {
+	return fmt.Sprintf(releaseNameTpl, kymaObjKey.Namespace, kymaObjKey.Name)
 }
 
-func prepareInstallInfo(ctx context.Context, chartPath, releaseName string, restConfig *rest.Config,
-	restClient client.Client, argsVals map[string]interface{},
-) moduleLib.InstallInfo {
-	return moduleLib.InstallInfo{
+func prepareInstallInfo(ctx context.Context, chartPath string, restConfig *rest.Config,
+	restClient client.Client, argsVals map[string]interface{}, kymaObjKey client.ObjectKey,
+) moduletypes.InstallInfo {
+	return moduletypes.InstallInfo{
 		Ctx: ctx,
-		ChartInfo: &moduleLib.ChartInfo{
+		ChartInfo: &moduletypes.ChartInfo{
 			ChartPath:   chartPath,
-			ReleaseName: releaseName,
-			Flags: moduleLibTypes.ChartFlags{
+			ReleaseName: skrChartReleaseName(kymaObjKey),
+			Flags: moduletypes.ChartFlags{
 				SetFlags: argsVals,
 			},
 		},
-		ClusterInfo: moduleLibTypes.ClusterInfo{
+		ResourceInfo: moduletypes.ResourceInfo{
+			BaseResource: watcherCachingBaseResource(kymaObjKey),
+		},
+		ClusterInfo: moduletypes.ClusterInfo{
 			Client: restClient,
 			Config: restConfig,
 		},
 	}
 }
+
+func watcherCachingBaseResource(kymaObjKey client.ObjectKey) *unstructured.Unstructured {
+	baseRes := &unstructured.Unstructured{}
+	baseRes.SetLabels(map[string]string{
+		modulelabels.CacheKey: kymaObjKey.String(),
+	})
+	baseRes.SetNamespace(metav1.NamespaceDefault)
+	baseRes.SetName(staticWatcherConfigName)
+	return baseRes
+}
+
+func CheckWebhookCABundleConsistency(ctx context.Context, skrClient client.Client, kymaObjKey client.ObjectKey) error {
```

Comment:
As far as I see `CheckWebhookCABundleConsistency` is only used in tests. Then why is it part of this file?